### PR TITLE
Verify passed dataset is a directory and not a file for classification training

### LINKED ISSUE #14732: How to export custom YOLOv8 classification model with INT8 calibration
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Currently, I am using Ultralytics version 8.2.18 and have tried to convert a custom YOLOv8 classification model from FP32 to INT8 using this command

`yolo export model=yolov8s_car_brand_fp32_25_07_2024.pt format=engine device=0 int8=True workspace=4 data=car_brand.yaml `

, but I got this error:

352481988-f517521c-21c5-40d6-ad61-70bc7be7946c

Inside car_brand.yaml, I have already defined the path and list of image files in .txt format for both the training set and test set as follows:

path: /home/nvidia/ORDS/int8/car_brand/yolov8
train: train_car_brand.txt
val: test_car_brand.txt

I am not sure how I should define content inside my custom yaml. Can you help me solve this export issue?

### Additional

_No response_

### LINKED ISSUE #16729: Does YOLO v8 classification not require dataset? I don't get it how to do train yolov8 classifier by custom dataset.
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hello. I am trying to train yolov8 classifier but I don't get it how to do. I only used YOLO for object detection, but not for classification. So I'm confused about preparing custom dataset. 

I saw solved question #1101 , and the answer was 'there's no need for dataset'. 
But if classification task doesn't need the dataset file, then how to specify number of classes and their names? 

When I did object detection, I labeled every objects for train set in coco dataset format. 
And now, I want to classify the document types by using YOLO. 

So here's my questions. 
1. Do I have to label all of my documents whether they are email or not? If so, how can I do? If not, then how can YOLO know the correct classes?
2. How can I make yaml file? There's no default YAML file in the ultralytics folder. 
3. How can I create train / val / test dataset for YOLO v8 classifier?

I need someone's help. 
Thank you so much. 

### Additional

_No response_

### LINKED ISSUE #18261: Error with yamal file in yolo11m-cls
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I distributed the data for yolo11m-cls according to the example:

image

This way each folder corresponds to the class of images in them.
But I don't know how to create a yaml file for this data form.

So how build yaml file for classification model?
I tried to use a yaml file for the detection model, but I got an error:

image

### Additional

_No response_

### PR DESCRIPTION
**Pull request by Y-T-G**
This has happened a couple of times already. Some users think that Classification training uses a YAML for dataset like other tasks, and the error message is not helpful.

https://www.reddit.com/r/computervision/comments/1nmuo7w/first_time_training_yolo_dataset_not_found/

Fixes https://github.com/ultralytics/ultralytics/issues/18261
Fixes https://github.com/ultralytics/ultralytics/issues/16729
Fixes https://github.com/ultralytics/ultralytics/issues/14732

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
Adds a clear validation to ensure classification datasets are directories, raising a helpful error instead of attempting a download. ✅

### 📊 Key Changes
- 🔍 In `check_cls_dataset`, if the provided classification dataset path looks like a file (has a suffix), a `ValueError` is raised with guidance.
- 📎 Error message includes a link to the classification dataset docs: [Classification dataset docs](https://docs.ultralytics.com/datasets/classify/).
- 🛑 Prevents unintended auto-download attempts when a file path is mistakenly supplied.

### 🎯 Purpose & Impact
- 🧭 Improves user experience with immediate, clear feedback when the dataset path is incorrect.
- ⏱️ Saves time and bandwidth by avoiding unnecessary downloads.
- 🔧 Helps users configure classification training correctly (expects a directory with the dataset structure).
- ⚠️ Potentially a breaking change for users who passed a file path (e.g., YAML); update scripts to provide a directory path.

Example:
```python
from ultralytics import YOLO

# Correct: directory path for classification datasets
model = YOLO('yolo11n-cls.pt')
model.train(data='path/to/classify_dataset')  # e.g., contains /train and /val subfolders
```

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Y-T-G, thank you for submitting an `ultralytics/ultralytics` 🚀 PR! This is an automated response to help streamline reviews—an Ultralytics engineer will follow up with you shortly.

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_ — Bruce Lee

Additional review notes to help move this forward:
- Please ensure the new validation logic in `check_cls_dataset` is covered by unit tests (e.g., passing a `.yaml` or `.zip` path vs. a valid directory) to prevent regressions. 🧪
- Confirm error message clarity and consistency with existing logging style, and ensure it won’t conflict with named-dataset download behavior.
- Double-check all CI jobs after pushing updates.

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Don’t hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🎯

**Comment by glenn-jocher** [comment]
@Y-T-G oh good check! Can you use `if data_dir.is_file():`?

**Comment by UltralyticsAssistant** [comment]
🎉 Fantastic merge! Huge thanks to @Y-T-G for the sharp validation update and to @glenn-jocher for the support. “Simplicity is the ultimate sophistication.” — Leonardo da Vinci. This PR embodies that—clear, immediate feedback that guides users to the right path and prevents accidental downloads, making classification training smoother across YOLO11 and beyond.

Quick reminder for users:
```python
from ultralytics import YOLO

model = YOLO('yolo11n-cls.pt')
model.train(data='path/to/classify_dataset')  # directory with /train and /val
```

For details, see the Classification dataset docs: https://docs.ultralytics.com/datasets/classify/

Appreciate the thoughtful improvement—small change, big UX win. Thank you! 🙌

**Comment by Y-T-G** [comment]
@glenn-jocher We can, but `is_file()` is False if the file doesn't exist, so it shows all the confusing attempting download and no valid images found error before it fails if the user passes a non-existent file. It's also slower:
```python
In [1]: from pathlib import Path

In [2]: test = Path("bad.yaml")

In [3]: %timeit test.is_file()
2.41 μs ± 2.82 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)

In [4]: %timeit test.suffix != ""
358 ns ± 0.347 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)
```
